enhancements/insights/pulling-sca-certs-from-ocm.md

```diff
@@ -0,0 +1,155 @@
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will enable the Insights Operator to pull the data (SCA certs)
+from the OCM (OpenShift Cluster Manager) API. The data will be exposed by the Insights Operator
+in the OpenShift API to allow users to use them when consuming and building container images
+on the platform.
+
+## Motivation
+
+Users could consume RHEL content and container images using the RHEL subscription in the OpenShift 3.x.
+In the OpenShift 4, this is no longer possible because the Red Hat Enterprise Linux Core OS (RHCOS) does not
+provide any attached subscription. This enhancement is to provide users the Simple Content Access (SCA) certs
+from Red Hat Subscription Manager (RHSM).
+
+### Goals
+
+- Extend the Insights Operator config with an OCM API URL to be able to query the data
+- Periodically pull the data from the OCM API and expose it in the OpenShift API
+- This is an opt-in feature by a cluster user and might be moved to a different OCP component in the future
+
+### Non-Goals
+
+- Insights Operator providing any transformation or post-processing of the SCA certs pulled
+  from the OCM API
+
+## Proposal
+
+### Why is it in the Insights Operator?
+
+The Insights Operator is now the only OCP component that connects an OpenShift cluster to a Red Hat subscription experience (console.redhat.com APIs). The consumers of the SCA certs are not only builds, but also shared resources, such as the CSI driver.
+
+### User Stories
+
+#### Consume SCA certs exposed in the API
+
+As an OpenShift user
+I want to consume SCA certs to be able to consume RHEL content and to build
+corresponding container images.
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: The Insights Operator is marked as Degraded (in case of HTTP error code other than 404, which means that the organization didn't allow this feature on the OCM side).
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the data from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config-managed` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### SCA certs in API
+
+The SCA certificate is available via the `etc-pki-entitlement` secret in the `openshift-config-managed` namespace. The secret will be available for use in other namespaces by creating a cluster-scoped `Share` resource. Cluster admin creates a `clusterrolebinding` to allow a service account access to the `Share` resource.
```

Comment:
I would propose we have the Insights Operator create the `Share` resource as well. This would help tell admins that the `etc-pki-entitlement` secret in the `openshift-config-managed` namespace is a candidate for propagation across namespaces. This was discussed initially but I had put this on hold since the `Share` API was under development as part of an OLM operator. Things have changed since the - we'll have the `Share` resource available on day 0.

I would suggest we add this to this EP and implement the same in parts whenever the `Share` resource becomes available on clusters.

Reply:
Based on our discussion, the creation of the `Share` resource object is described in the [### Tech Preview -> GA](https://github.com/openshift/enhancements/pull/683/files#diff-93dffd2aa59c973a916306675f62907275cff29ff9a16ba77aaf98cfa830249fR132)